src/fidesops/service/connectors/query_config.py

```diff
@@ -694,56 +695,83 @@ def get_request_by_action(self, action: str) -> SaaSRequest:
+
+        filtered_data = self.node.typed_filtered_values(input_data)
+
+        # populate the SaaS request with reference values from other datasets provided to this node
+        request_params = []
+        for string_path, reference_values in filtered_data.items():
+            for value in reference_values:
+                request_params.append(
+                    self.generate_query({string_path: [value]}, policy)
+                )
+        return request_params
+
+    def generate_query(
+        self, input_data: Dict[str, List[Any]], policy: Optional[Policy]
     ) -> SaaSRequestParams:
         """
-        Populates the placeholders in the request with the given param values
+        This returns the query/path params needed to make an API call.
+        This is the API equivalent of building the components of a database
+        query statement (select statement, where clause, limit, offset, etc.)
         """
-        path = request.path
+        current_request = self.get_request_by_action("read")
+
+        path: str = current_request.path
         params: Dict[str, Any] = {}
-        data: Dict[str, Any] = {}
 
-        for param in request.request_params:
+        # uses the param names to read from the input data
+        for param in current_request.request_params:
             if param.type == "query":
                 if param.default_value:
                     params[param.name] = param.default_value
                 elif param.references or param.identity:
-                    params[param.name] = param_values[param.name]
+                    params[param.name] = input_data[param.name][0]
             elif param.type == "path":
-                path = path.replace(f"<{param.name}>", param_values[param.name])
-            elif param.type == "body":
-                data[param.name] = param_values[param.name]
+                path = path.replace(f"<{param.name}>", input_data[param.name][0])
 
-        return path, params, data
+        logger.info(f"Populated request params for {current_request.path}")
+        return "GET", path, params, None
 
-    def generate_query(
-        self, input_data: Dict[str, List[Any]], policy: Optional[Policy]
-    ) -> Optional[List[SaaSRequestParams]]:
+    def generate_update_stmt(
+        self, row: Row, policy: Policy, request: PrivacyRequest
+    ) -> SaaSRequestParams:
         """
-        This returns the query/path params needed to make an API call.
-        This is the API equivalent of building the components of a database
-        query statement (select statement, where clause, limit, offset, etc.)
+        Prepares the update request by masking the fields in the row data based on the policy.
+        This masked row is then added as the body to a dynamically generated SaaS request.
         """
+        update_value_map: Dict[str, Any] = self.update_value_map(row, policy, request)
+        body: Row = unflatten_dict(update_value_map)
```

Comment:
Just since we're here, can you move both `update_value_map` and `body` down to be right above your return statement? `return "PUT", path, params, json.dumps(body)`.  Defining them near where you're using them will help separate that they are only related to body, you're using the original row for building the query and the path.

Reply:
Makes sense to keep them together, done!